Strict host header validation breaks route extraction

### Command

build

### Is this a regression?

- [x] Yes, this behavior used to work in the previous version

### The previous version in which this bug was not present was

20.3.13

### Description

Since an update of @angular/cli from `20.3.13` to `20.3.19`, "extracting routes" fails when running `ng build`. The route extraction procedure boots an Angular app wherein my `server.ts` file requires certain `process.env` keys to exist, and otherwise throws an error. These variables used to be inherited from the parent process but no longer are.

I have pinpointed the issue to https://github.com/angular/angular-cli/pull/32516, where in `prerender.ts` new keys `env` are set which changes the behavior of [lines 33-39 of WorkerPool](https://github.com/angular/angular-cli/blob/598a690a0baea5cd54af5cea38e673c33605a627/packages/angular/build/src/utils/worker-pool.ts#L33-L39) that now hits the `if` branch instead of the `else` branch.

I have confirmed this is the case by changing the blocks in prerender.ts as follows:
```diff
    env: {
++    ...process.env,
      'NG_ALLOWED_HOSTS': 'localhost',
    },
```

Is this intended behavior? If so, how do I fix it given that all of this looks to be internal angular builder logic?

Thanks!

### Minimal Reproduction

Add a block to server.ts:

```typescript
if (! process.env.TEST) {
  throw new Error('oops');
}
```

run the following command:

```
TEST=true npx ng build
```

### Exception or Error

```text
✘ [ERROR] An error occurred while extracting routes.

oops
```

### Your Environment

```text
_                      _                 ____ _     ___
    / \   _ __   __ _ _   _| | __ _ _ __     / ___| |   |_ _|
   / △ \ | '_ \ / _` | | | | |/ _` | '__|   | |   | |    | |
  / ___ \| | | | (_| | |_| | | (_| | |      | |___| |___ | |
 /_/   \_\_| |_|\__, |\__,_|_|\__,_|_|       \____|_____|___|
                |___/



Angular CLI: 20.3.19
Node: 24.3.0
Package Manager: npm 11.4.2
OS: darwin arm64


Angular: 20.3.17
... animations, common, compiler, compiler-cli, core, forms
... platform-browser, platform-browser-dynamic, platform-server
... router

Package                         Version
---------------------------------------
@angular-devkit/architect       0.2003.19
@angular-devkit/build-angular   20.3.19
@angular-devkit/core            20.3.19
@angular-devkit/schematics      20.3.19
@angular/build                  20.3.19
@angular/cdk                    20.2.4
@angular/cli                    20.3.19
@angular/material               20.2.4
@angular/ssr                    20.3.19
@schematics/angular             20.3.19
rxjs                            7.8.2
typescript                      5.9.2
zone.js                         0.15.1
```

### Anything else relevant?

_No response_

--- a/packages/angular/build/src/utils/server-rendering/prerender.ts
+++ b/packages/angular/build/src/utils/server-rendering/prerender.ts
@@ -226,6 +226,7 @@ async function renderPages(
     } as RenderWorkerData,
     execArgv: workerExecArgv,
     env: {
+      ...process.env,
       'NG_ALLOWED_HOSTS': 'localhost',
     },
   });
@@ -341,6 +342,7 @@ async function getAllRoutes(
     } as RoutesExtractorWorkerData,
     execArgv: workerExecArgv,
     env: {
+      ...process.env,
       'NG_ALLOWED_HOSTS': 'localhost',
     },
   });

